CHANGELOG.md

```diff
@@ -26,6 +26,11 @@ Dropping a requirement of a major version of a dependency is a new contract.
 ## [Unreleased]
```

Comment:
One last thing: let's clean up the Git history.
In this case it looks like an atomic feature, so we can squash it all into one commit. Force-pushing branches in order to preserve Git hygiene is fine and even recommended.

Reply:
Generally, I would keep commits separated for the sake of code review. But since this is reviewed & approved now, agreed.